packages/lab/src/badge/Badge.tsx

```diff
@@ -60,32 +34,39 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
     window: targetWindow,
   });
 
-  const badgeId = useId();
-  const childId = useId(
-    isValidElement<HTMLAttributes<HTMLElement>>(children)
-      ? children.props?.id
-      : undefined
-  );
+  let valueText = value;
 
-  let badgeContentValue = badgeContent;
-  if (badgeContentValue > max) {
-    badgeContentValue = `${max}+`;
+  if (typeof value === "number" && max) {
+    valueText = value > max ? `${max}+` : value;
+  } else if (typeof value === "string") {
+    valueText = value.length > 4 ? `${value.slice(0, 4)}...` : value;
   }
 
-  return (
-    <span
-      className={clsx(withBaseName(), className)}
-      ref={ref}
-      role="img"
-      aria-labelledby={clsx(childId, badgeId)}
-      {...rest}
-    >
-      {children && isValidElement<HTMLAttributes<HTMLElement>>(children)
-        ? cloneElement(children, { id: childId })
-        : children}
-      <span id={badgeId} className={clsx(withBaseName("badge"))}>
-        {badgeContentValue}
+  if (!children) {
+    return (
+      <div>
```

Comment:
You don't need this div. Make sure to create a storybook example where you are rendering the badge inline with some text and no `display: flex` on the parent. You will  immediately notice the issue. Divs have `display: block` by default meaning they will take up the whole line.

```tsx
<span>some text <Badge value="hello" /></span>
```

Reply:
When rendering inline, you can simply return the <span/> below directly. No need for a wraper.